erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -923,6 +923,12 @@ frappe.ui.form.on('Payment Entry', {
 				}
 			});
 		}
+	},
+
+	validate:function(frm) {
+		if (frm.doc.references.length === 0) {
```

Comment:
Why so? Reference is not mandatory in payment entry for advance payment

Reply:
Then we have to handle the scenario where PO is created with on_hold flag and PE is created against that.

Reply:
Raise a separate PR for that scenario